components/territorio/cursos/TS-Cursos.tsx

```diff
@@ -0,0 +1,74 @@
+import type { MediaQueryKey } from 'deco-sites/territorio/hooks/useTsMediaQuery.tsx';
+import useTsMediaQuery from 'deco-sites/territorio/hooks/useTsMediaQuery.tsx';
+import { AppContext } from '../../../apps/site.ts';
+import Icon from '../../../components/ui/Icon.tsx';
+import { useTsCarrousel } from '../../../hooks/useTsCarrousel.tsx';
+import type { Course } from './TS-CursoCard.tsx';
+import TsCursoCard from './TS-CursoCard.tsx';
+
+export interface CursosProps {
+  courses: Course[];
+}
+
+const DISPLAYED_ITEMS: Record<MediaQueryKey, number> = {
+  sm: 2,
+  md: 2,
+  lg: 2,
+  xl: 3,
+  '2xl': 4,
+};
+
+const TsCursos = ({ courses }: CursosProps) => {
+  const { currentMediaQuery } = useTsMediaQuery();
+
+  const { visibleItems, onNext, onPrevious } = useTsCarrousel({
+    items: courses,
+    visibleItemsCountParam: DISPLAYED_ITEMS[currentMediaQuery],
```

Comment:
Suggestion (non-blocking): You can pass the object directly:

```suggestion
    visibleItemsCountParam: DISPLAYED_ITEMS,
```
So you don't need to use the media query hook for that

Reply:
Nice!